modules/tensor_mechanics/src/materials/abaqus/AbaqusUMATStress.C

```diff
@@ -287,18 +301,30 @@ AbaqusUMATStress::computeQpStress()
   _stress[_qp] = RankTwoTensor(
       _aqSTRESS[0], _aqSTRESS[1], _aqSTRESS[2], _aqSTRESS[5], _aqSTRESS[4], _aqSTRESS[3]);
 
+  // check entries for NaN
+  if (_check_nan)
+    for (const auto i : make_range(6))
+      if (std::isnan(_aqSTRESS[i]))
+        mooseError(
+            "The UMAT plugin in AbaqusUMATStress '",
+            name(),
+            "' returned NaNs in the computed stress and the MOOSE simulation is not using an "
+            "incremental/finite strain formulation. This likely means that the used UMAT plugin is "
```

Comment:
incremental/finite strain means 'incremental finite strain' or 'incremental or finite strain'? (For user's sake)